test/gtest/kerasmomentum.hpp

```diff
@@ -0,0 +1,203 @@
+#include <miopen/miopen.h>
+
+template <class T>
+inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v)
+{
+    os << '{';
+    for(int i = 0; i < v.size(); ++i)
+    {
+        if(i != 0)
+            os << ',';
+        os << v[i];
+    }
+    os << '}';
+    return os;
+}
+
+struct KerasMomentumTestCase
+{
+    std::vector<size_t> dims;
+    bool nesterov      = false;
+    bool is_contiguous = true;
+
+    friend std::ostream& operator<<(std::ostream& os, const KerasMomentumTestCase& tc)
+    {
+        return os << " dims:" << tc.dims << "nesterov" << tc.nesterov
+                  << " is_contiguous:" << tc.is_contiguous;
+    }
+
+    std::vector<size_t> ComputeStrides(std::vector<size_t> inputDim) const
+    {
+        if(!is_contiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!is_contiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
+inline std::vector<KerasMomentumTestCase> KerasMomentumTestConfigs()
+{ // n c d h w lr momentum dampening weightDecay nesterov momentumInitialized
+    return {
+        {{50, 10}, true, true},
+        {{50, 10}, false, true},
+        {{50, 10, 20}, true, true},
+        {{50, 10, 20}, false, true},
+        {{50, 10, 20, 30}, true, true},
+        {{50, 10, 20, 30}, false, true},
+        {{50, 10, 20, 30, 4}, true, true},
+        {{50, 10, 20, 30, 4}, false, true},
+        {{50, 10}, true, false},
+        {{50, 10}, false, false},
+        {{50, 10, 20}, true, false},
+        {{50, 10, 20}, false, false},
+        {{50, 10, 20, 30}, true, false},
+        {{50, 10, 20, 30}, false, false},
+        {{50, 10, 20, 30, 4}, true, false},
+        {{50, 10, 20, 30, 4}, false, false},
+    };
+}
+
+template <typename T = float>
+struct KerasMomentumTest : public ::testing::TestWithParam<KerasMomentumTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle        = get_handle();
+        KerasMomentum_config = GetParam();
+
+        auto gen_var   = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1, 10); };
+        auto gen_accum = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1, 100); };
+        auto gen_lr    = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1, 50); };
+        auto gen_grad  = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1, 20); };
+        auto gen_mmt   = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1, 10); };
+
+        auto dims    = KerasMomentum_config.dims;
+        auto strides = KerasMomentum_config.ComputeStrides(dims);
+        nesterov     = KerasMomentum_config.nesterov;
+
+        var_in      = tensor<T>{dims, strides}.generate(gen_var);
+        var_out     = tensor<T>{dims};
```

Comment:
why you don't pass `strides` at here

Reply:
I only make input non-contiguous at Modnn and Rocm test